src/MonoTorrent/MonoTorrent.Client/Managers/TorrentManager.cs

```diff
@@ -286,6 +288,48 @@ public class TorrentManager : IDisposable, IEquatable<TorrentManager>
 
         public InfoHash InfoHash { get; }
 
+        /// <summary>
+        /// Torrent tracker announce Urls
+        /// </summary>
+        public IList<RawTrackerTier> AnnounceUrls
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Magnet link
+        /// </summary>
+        public MagnetLink MagnetLink {
+            get {
+                if (this.magnetLink == null) {
+                    this.magnetLink = new MagnetLink (Torrent.InfoHash, name: Name,
```

Comment:
Couldn't we just invoke `Torrent.ToMagnetLink` rather than reimplementing it? This method does rely on `Torrent` being non-null so it should simpler!